src/utils/ao.ts

```diff
@@ -171,22 +172,86 @@ export async function spawnANT({
     }
   }
 
-  // for hyperbeam caching, due to a SU issue, we need to send a second message to the ANT to cache the state
+  // Note: for hyperbeam caching, due to a SU issue, we need to send a second message to the ANT to cache the state
   // We wait for the first message to be processed before sending the second one to ensure this is the second message
-  const processApi = new AOProcess({
-    processId,
-    ao,
-    logger,
-  });
-  await processApi.send({
-    tags: [{ name: 'Action', value: 'State' }],
-    signer,
-  });
+  // We use the resulting state to check the owner of the ANT is set in the registry. Should this be patched on MUs,
+  // we can convert to just a simple dry-run to avoid sending/signing another message.
+  let owner: WalletAddress | undefined;
+  try {
+    const processApi = new AOProcess({
+      processId,
+      ao,
+      logger,
+    });
+    const { id } = await processApi.send({
+      tags: [{ name: 'Action', value: 'State' }],
+      signer,
+    });
+    const stateResult = await ao.result({
+      process: processId,
+      message: id,
+    });
+    if (stateResult === undefined) {
+      throw new Error('Failed to get state result');
+    }
+    const { Owner } = JSON.parse(stateResult.Messages?.[0]?.Data ?? '{}') as {
+      Owner: WalletAddress;
+    };
+    owner = Owner;
+    logger.debug(`Successfully spawned new ANT and validated owner`, {
+      processId,
+      module,
+      owner,
+    });
+  } catch (error) {
+    logger.error('Failed to validate owner of spawned ANT', {
+      processId,
+      module,
+      error,
+    });
+    throw error;
+  }
 
-  logger.debug(`Successfully spawned new ANT`, {
-    processId,
-    module,
-  });
+  /**
+   * Now confirm the owner of the ANT is set in the registry
+   * This is to ensure the ANT is available via the ANT registry
+   * for the owner to find and use the ANT.
+   */
+  if (owner !== undefined) {
```

Comment:
We should always check the acl here - so if owner is undefined, thats an error path (throw).

Can change this to
```
if (owner === undefined) throw new Error("...")
```

Reply:
https://github.com/ar-io/ar-io-sdk/commit/85460ae7a3ce73cc04ff22f63c273ec06dd7be2b